scripts/tdr/copy_from_tdr_to_gcs_hca/copy_from_tdr_to_gcs_hca.py

```diff
@@ -0,0 +1,216 @@
+# Description: This script copies data from Terra Data Repository (TDR) HCA project buckets to HCA staging area buckets.
+# It is based on the bash script get_snapshot_files_and_transfer.sh, written by Samantha Velasquez.
+
+import os
+import sys
+import csv
+import logging
+import re
+import requests
+import google.auth
+import google.auth.transport.requests
+import subprocess
+
+STAGING_AREA_BUCKETS = {
+        "EBI": "gs://broad-dsp-monster-hca-prod-ebi-storage/prod",
+        "UCSC": "gs://broad-dsp-monster-hca-prod-ebi-storage/prod",
+        "LANTERN": "gs://broad-dsp-monster-hca-prod-lantern",
+        "LATTICE": "gs://broad-dsp-monster-hca-prod-lattice/staging",
+        "TEST": "gs://broad-dsp-monster-hca-prod-ebi-storage/broad_test_dataset"
+}
+
+
+def setup_cli_logging_format() -> None:
+    logging.basicConfig(level=logging.INFO, format='%(message)s', stream=sys.stdout)
+
+
+def validate_input(csv_path: str):
```

Comment:
Do we want to validate headers + number of columns too?

Reply:
This is generic csv validator that Sam wrote in her [original bash script.](https://github.com/broadinstitute/horsefish/blob/FE-287-implement-copy-from-tdr-to-gcs/scripts/tdr/copy_from_tdr_to_gcs_hca/get_snapshot_files_and_transfer.sh)

In _parse_csv() which is specific to this HCA manifest, [I validate the number of columns](https://github.com/broadinstitute/horsefish/blob/4079e7f641dbd9a532cd9f7f1fd630b2618d8b52/scripts/tdr/copy_from_tdr_to_gcs_hca/copy_from_tdr_to_gcs_hca.py#L82C13-L82C33) and if there is some odd row in there - it won't pass muster for the institution check.

```
assert len(row) == 2
            institution = row[0]
            project_id = find_project_id_in_str(row[1])

            if institution not in STAGING_AREA_BUCKETS:
                raise Exception(f"Unknown institution {institution} found. "
                                f"Make sure the institution is in the list of staging area buckets and is in all caps")
```

Reply:
added as a possible improvement in the readme